### DISCUSSION TITLE
Validation not working consistently for useFieldArray

### DISCUSSION THREAD

**Discussion by MartinCerny-awin** [react-hook-form/react-hook-form#2156] [Q&A] [answered]
I am having form using useFieldArray, where the first field is returning an object. I am using it as a select in my form. After you select an option the object is passed to onChange and stored to the array.

Everything works perfectly, but the validation for the first object is not getting triggered. The same field is getting validated in the second object.

https://codesandbox.io/s/suspicious-hertz-uy4g4

**Comment by bluebill1049** [comment, marked as answer]
hey @marsinearth make sure use object as `useFieldArray`, because the field will need to be injected with `id`

<img width="916" alt="Screen Shot 2020-07-11 at 10 00 03 am" src="https://user-images.githubusercontent.com/10513364/87211537-4c6fed80-c35d-11ea-9699-fc996dde5d87.png">

**Reply by marsinearth** [reply]
you're keep mentioning wrong guy! haha..

**Reply by bluebill1049** [reply]
oops sorry 🙏  @marsinearth -__-
